### PR TITLE
Customize Contacts Waiting channels

### PR DESCRIPTION
**Pull request by GPaoloni**
Primary reviewer: @murilovmachado 

## Description
This PR:
- Adds the option to customize the channels displayed in contacts waiting widget. This is done via a new attribute in service configuration: `contactsWaitingChannels` (typed `contactsWaitingChannels: ChannelTypes[]`, i.e. a list of channel types). If `contactsWaitingChannels` is missing in service config, all channels will be shown. If `contactsWaitingChannels` is present, it must be a list with the lowercase names of the channels the helpline wants to display (e.g. `['voice', 'web', 'twitter']`). @dee-luo @Humairaa127  [here](https://github.com/techmatters/flex-plugins/blob/ce03aeceb617da3b3bcd64dab4b23ff2918ab420/plugin-hrm-form/src/states/DomainConstants.ts#L1) you can see the valid ChannelTypes.
- Refined some types in `HrmFormPlugin.tsx`.
- Moved `QueuesStatus` to TS.
- Moved `QueueCard` to TS and Hooks API.
- Reworked `QueuesStatus` tests to use testing library.

Notes:
- In this context customize the channels means you can choose which channels to display. The internal subscription to the channels remains the same, as I don't see any benefit on disabling the subscriptions for channels that will likely not be wired to Flex anyways

### Checklist
- [ ] [Corresponding issue has been opened](https://bugs.benetech.org/browse/CHI-1085)
- [ ] New tests added
- [N/A] Strings are localized
- [ ] Tested for chat contacts
- [ ] Tested for call contacts

### Verification steps
- `npm run dev` should show the app as usual.
- Add `contactsWaitingChannels` (can be done via `getConfig`, in `HrmFormPlugin.tsx`).
- Check that now only the channels listed in `contactsWaitingChannels` are shown.

### CODE REVIEW

File: plugin-hrm-form/src/states/DomainConstants.ts [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
-  facebook: 'facebook',
-  web: 'web',
   voice: 'voice',
   sms: 'sms',
+  facebook: 'facebook',
   whatsapp: 'whatsapp',
+  web: 'web',
   twitter: 'twitter',
   instagram: 'instagram',
 } as const;
```
Comment (GPaoloni) [inline plugin-hrm-form/src/states/DomainConstants.ts:9]: Sorted this so Object.values preserves the desired order.

File: plugin-hrm-form/src/components/queuesStatus/QueueCard.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
+/* eslint-disable react/prop-types */
```
Comment (GPaoloni) [inline plugin-hrm-form/src/components/queuesStatus/QueueCard.tsx:1]: I've used `git mv` here, don't know why it shows the whole diff :/

**Comment by GPaoloni** [comment]
Looks like installing `@types/jest-axe` dependency in my machine causes some issues with the CI (CI failing in https://github.com/techmatters/flex-plugins/pull/708/commits/bf237c983642993a78971e6e209ac1e9e1e1aa85, reverted in https://github.com/techmatters/flex-plugins/pull/708/commits/730190f34a9ca734ffed7bd14f898a9f66bc2f1a). Can this be due to the [CI using Node v12](https://github.com/techmatters/flex-plugins/blob/ce03aeceb617da3b3bcd64dab4b23ff2918ab420/.github/workflows/plugin-hrm-form-ci.yml#L12)? I think is an issue related to npm, if I install the dependency with my local setup `package-lock.json` gets lots of changes (you can see in the failing commit).

**Comment by GPaoloni** [comment]
Continuing on the above comment, I've created a [new branch](https://github.com/techmatters/flex-plugins/tree/gian_test-npm) with nothing but `npm install` (Node v14.18.1 & npm v8.3.2) and the CI is failing in the same way. Looks like some conflict with the `package-lock.json`?

**Review by murilovmachado — APPROVED** [review]
Behavior and code LGTM, thanks for improving our typings as well.

**Comment by murilovmachado** [comment]
Also, I've downloaded your branch and `UNBUNDLED_REACT=true npm ci` worked fine for me. Didn't get any issues.
node v14.18.2
npm v6.14.15

**Comment by GPaoloni** [comment]
>Also, I've downloaded your branch and UNBUNDLED_REACT=true npm ci worked fine for me. Didn't get any issues.
node v14.18.2
npm v6.14.15

@murilovmachado Yes it works locally, but fails the CI when code is pushed :/

**Pull request merged**